llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -483,88 +484,85 @@ std::vector<uint32_t> getKernelReqdWorkGroupSizeMetadata(const Function &Func) {
   return {X, Y, Z};
 }
 
-// Input parameter KernelModuleMap is a map containing groups of entry points
-// with same values of the sycl-module-id attribute. ResSymbolsLists is a vector
-// of entry points names lists. Each vector element is a string with entry point
-// names from the same module separated by \n.
-// The function saves names of entry points from one group to a single
-// std::string and stores this string to the ResSymbolsLists vector.
-void collectSymbolsLists(
-    const std::map<StringRef, std::vector<const Function *>> &KernelModuleMap,
-    string_vector &ResSymbolsLists) {
-  for (const auto &It : KernelModuleMap) {
-    std::string SymbolsList;
-    for (const auto &F : It.second) {
-      SymbolsList =
-          (Twine(SymbolsList) + Twine(F->getName()) + Twine("\n")).str();
+// Input parameter ModuleEntryPoints contains group of entry points with same
+// values of the sycl-module-id attribute. ResSymbolsStr is a string with entry
+// point names from the same module separated by \n.
+std::string collectSymbolsList(const FuncPtrVector *ModuleEntryPoints) {
```

Comment:
> Yes, it may be equal to nullptr in case if GlobalsSet is empty, which in turn means that there are no entry points in input module. I'll update a comment for this function.

Why can't we create an empty vector if `GlobalsSet` is empty and simply pass reference to it everywhere?

Reply:
Added empty vector in case if `GlobalsSet` is empty.